sdks/python/apache_beam/testing/synthetic_pipeline.py

```diff
@@ -61,6 +65,35 @@
   np = None
 
 
+class _Random(Random):
+  """A subclass of `random.Random` from the Python Standard Library that
+  provides a method returning random bytes of arbitrary length.
+  """
+
+  # `numpy.random.RandomState` does not provide `random()` method, we keep this
+  # for compatibility reasons.
+  random_sample = Random.random
+
+  def bytes(self, length):
+    """Returns random bytes.
+
+    Args:
+      length (int): Number of random bytes.
+    """
+    n = length // 8 + 1
+    # pylint: disable=map-builtin-not-iterating
+    return struct.pack(
+        '{}Q'.format(n),
+        *map(self.getrandbits, itertools.repeat(64, n)))[:length]
```

Comment:
Since we don't need py2 compatibility anymore, consider using `to_bytes`. Here's an equivalent for chunk_size=8. It seems to be somewhat slower than current method (perhaps since I'm not using `map+repeat()`), but with larger `chuck_size`, seems to be more efficient. Large chunk size may be less efficient for short bytesequences.

```
chunk_size_bytes = 8 // TBD - larger chunks seem to improve performance.
chunk_size_bits = chunk_size_bytes * 8
num_chunks = length // chunk_size_bytes + 1

return b''.join([self.getrandbits(chunk_size_bits).to_bytes(chunk_size_bytes, sys.byteorder) for _ in range(num_chunks)])[:length]
```

If you decide to keep current implementation - please add a comment explaining the mechanics for readers not familiar with this code (we generate 8-byte stings, and then fit them into a representation of C++'s long-long, which also takes up 8 bytes).

Reply:
How about not using chunks at all? I did some tests again and, surprisingly, it looks like we don't need them. 
Here's a test with a set of different chunk sizes:
`for CHUNK_SIZE in {4,8,32,64}; do python -m timeit -s "import random; import sys; chunk_size=$CHUNK_SIZE; len=10; num_chunks=len//chunk_size+1" 'b"".join([random.getrandbits(chunk_size * 8).to_bytes(chunk_size, sys.byteorder) for _ in range(num_chunks)])[:len]'; done
`

Results:
for len==10:
200000 loops, best of 5: 1.62 usec per loop
200000 loops, best of 5: 1.34 usec per loop
200000 loops, best of 5: 1.02 usec per loop
200000 loops, best of 5: 1.19 usec per loop

for len==1000:
5000 loops, best of 5: 87.7 usec per loop
5000 loops, best of 5: 50.7 usec per loop
20000 loops, best of 5: 16.7 usec per loop
20000 loops, best of 5: 11 usec per loop

And without chunks:
`python -m timeit -s "import random; import sys; len=10" 'random.getrandbits(len * 8).to_bytes(len, sys.byteorder)'`

for len==10:
1000000 loops, best of 5: 358 nsec per loop

for len==1000:
50000 loops, best of 5: 4.5 usec per loop

Reply:
I'll push a commit with that so you can see how it looks in the code.

Reply:
This is great. I really like how we simplified this code.